source/source_integration_test.go

```diff
@@ -211,6 +216,95 @@ func TestSource_Snapshot_Continue(t *testing.T) {
 	is.NoErr(err)
 }
 
+func TestSource_CDC_Success(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	ctx := context.Background()
+
+	tableName := "conduit_source_cdc_integration_test_success"
+
+	cfg := prepareConfig(t, tableName)
+	cfg[ConfigKeyColumns] = "int_column,text_column"
+
+	err := prepareData(
+		ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, true,
+	)
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = clearData(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName)
+		is.NoErr(err)
+	})
+
+	s := new(Source)
+
+	err = s.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	// wait a bit for Vitess to get the "create table" query results from each shard
+	time.Sleep(time.Second * 3)
+
+	// Start first time with nil position.
+	err = s.Open(ctx, nil)
+	is.NoErr(err)
+
+	// switch to CDC iterator
+	_, err = s.Read(ctx)
+	is.Equal(err, sdk.ErrBackoffRetry)
+
+	err = insertRow(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, 1, "Bob")
+	is.NoErr(err)
+
+	time.Sleep(time.Second * 5)
+
+	readCtx, cancel := context.WithTimeout(ctx, time.Second*15)
+	defer cancel()
+
+	record, err := readWithRetry(readCtx, t, s, time.Second*5)
+	is.NoErr(err)
+	is.Equal(record.Metadata["action"], "insert")
+	is.Equal(record.Key.Bytes(), []byte(`{"int_column":1}`))
+	is.Equal(record.Payload.Bytes(), []byte(`{"int_column":1,"text_column":"Bob"}`))
+
+	err = updateRow(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, 1, "Alex")
+	is.NoErr(err)
+
+	record, err = readWithRetry(readCtx, t, s, time.Second*5)
+	is.NoErr(err)
+	is.Equal(record.Metadata["action"], "update")
+	is.Equal(record.Key.Bytes(), []byte(`{"int_column":1}`))
+	is.Equal(record.Payload.Bytes(), []byte(`{"int_column":1,"text_column":"Alex"}`))
+
+	err = deleteRow(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, 1)
+	is.NoErr(err)
+
+	record, err = readWithRetry(readCtx, t, s, time.Second*5)
+	is.NoErr(err)
+	is.Equal(record.Metadata["action"], "delete")
+	is.Equal(record.Key.Bytes(), []byte(`{"int_column":1}`))
+
+	err = s.Teardown(ctx)
+	is.NoErr(err)
+
+	err = insertRow(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, 2, "Ben")
+	is.NoErr(err)
+
+	// Open from the previous position.
+	err = s.Open(ctx, record.Position)
+	is.NoErr(err)
+
+	record, err = readWithRetry(readCtx, t, s, time.Second*5)
+	is.NoErr(err)
+	is.Equal(record.Metadata["action"], "insert")
+	is.Equal(record.Key.Bytes(), []byte(`{"int_column":2}`))
+	is.Equal(record.Payload.Bytes(), []byte(`{"int_column":2,"text_column":"Ben"}`))
+
+	err = s.Teardown(ctx)
+	is.NoErr(err)
+}
+
```

Comment:
could you please add test for cdc from empty table

Reply:
Added